go/store/types/value_store.go

```diff
@@ -460,101 +460,99 @@ func (lvs *ValueStore) Rebase(ctx context.Context) error {
-				return err
-			}
+	put := func(h hash.Hash, chunk chunks.Chunk) error {
+		err := lvs.cs.Put(ctx, chunk)
 
-			delete(lvs.bufferedChunks, h)
-			delete(lvs.withBufferedChildren, h)
-			lvs.bufferedChunkSize -= uint64(len(chunk.Data()))
-			return nil
+		if err != nil {
+			return err
 		}
 
-		for parent := range lvs.withBufferedChildren {
-			if pending, present := lvs.bufferedChunks[parent]; present {
-				err := WalkRefs(pending, lvs.nbf, func(reachable Ref) error {
-					if pending, present := lvs.bufferedChunks[reachable.TargetHash()]; present {
-						return put(reachable.TargetHash(), pending)
-					}
-
-					return nil
-				})
+		delete(lvs.bufferedChunks, h)
+		delete(lvs.withBufferedChildren, h)
+		lvs.bufferedChunkSize -= uint64(len(chunk.Data()))
+		return nil
+	}
 
-				if err != nil {
-					return false, err
+	for parent := range lvs.withBufferedChildren {
+		if pending, present := lvs.bufferedChunks[parent]; present {
+			err := WalkRefs(pending, lvs.nbf, func(reachable Ref) error {
+				if pending, present := lvs.bufferedChunks[reachable.TargetHash()]; present {
+					return put(reachable.TargetHash(), pending)
 				}
 
-				err = put(parent, pending)
-
-				if err != nil {
-					return false, err
-				}
-			}
-		}
-		for _, c := range lvs.bufferedChunks {
-			// Can't use put() because it's wrong to delete from a lvs.bufferedChunks while iterating it.
-			err := lvs.cs.Put(ctx, c)
+				return nil
+			})
 
 			if err != nil {
 				return false, err
 			}
 
-			lvs.bufferedChunkSize -= uint64(len(c.Data()))
-		}
-
-		d.PanicIfFalse(lvs.bufferedChunkSize == 0)
-		lvs.withBufferedChildren = map[hash.Hash]uint64{}
-		lvs.bufferedChunks = map[hash.Hash]chunks.Chunk{}
-
-		if lvs.enforceCompleteness {
-			root, err := lvs.Root(ctx)
+			err = put(parent, pending)
 
 			if err != nil {
 				return false, err
 			}
+		}
+	}
+	for _, c := range lvs.bufferedChunks {
+		// Can't use put() because it's wrong to delete from a lvs.bufferedChunks while iterating it.
+		err := lvs.cs.Put(ctx, c)
 
-			if (current != hash.Hash{} && current != root) {
-				if _, ok := lvs.bufferedChunks[current]; !ok {
-					// If the client is attempting to move the root and the referenced
-					// value isn't still buffered, we need to ensure that it is contained
-					// in the ChunkStore.
-					lvs.unresolvedRefs.Insert(current)
-				}
-			}
-
-			PanicIfDangling(ctx, lvs.unresolvedRefs, lvs.cs)
+		if err != nil {
+			return false, err
 		}
 
-		success, err := lvs.cs.Commit(ctx, current, last)
+		lvs.bufferedChunkSize -= uint64(len(c.Data()))
+	}
+
+	d.PanicIfFalse(lvs.bufferedChunkSize == 0)
+	lvs.withBufferedChildren = map[hash.Hash]uint64{}
+	lvs.bufferedChunks = map[hash.Hash]chunks.Chunk{}
+
+	if lvs.enforceCompleteness {
+		root, err := lvs.Root(ctx)
 
 		if err != nil {
 			return false, err
 		}
 
-		if !success {
-			return false, nil
+		if (current != hash.Hash{} && current != root) {
+			if _, ok := lvs.bufferedChunks[current]; !ok {
```

Comment:
What's happening here? Isn't `lvs.bufferedChunks` necessarily empty here? Where we supposed to check this somewhere else / further up in the function?

Reply:
I didn't change this, just removed the anonymous func() invocation it was wrapped in for no reason.

Very possible this has bugs, but they were pre-existing conditions.